src/lob-address-elements.js

```diff
@@ -1,527 +1,724 @@
-                            } else {
-                                cb(null);
-                            }
-                        }
-                    }
-                    xhr.send(JSON.stringify({
-                        address_prefix: query,
-                        city: config.elements.city.val(),
-                        zip_code: config.elements.zip.val(),
-                        state: config.elements.state.val(),
-                        geo_ip_sort: true
-                    }));
-                }
-                return false;
+        function denormalizeParts(data, bSecondary) {
+          var sd = data.components.secondary_designator;
+          if (data.secondary_line || config.denormalize === false) {
+            //echo exactly when configured explicitly or structurally required
+            return {
+              secondary_line: data.secondary_line,
+              primary_line: data.primary_line
             }
+          } else if (sd && bSecondary) {
+            //the user entered the value in the secondary line; echo there
+            var parts = data.primary_line.split(sd);
+            return {
+              secondary_line: sd + ' ' + (parts[1]).trim(),
+              primary_line: (parts[0]).trim()
+            }
+          } else {
+            //use the default
+            return {
+              secondary_line: data.secondary_line,
+              primary_line: data.primary_line
+            }
+          }
+        }
 
-            /**
-             * Project the chosen suggested address into the UI
-             * @param {object} suggestion - as returned from the Lob API
-             */
-            config.do.apply = function (suggestion) {
-                config.elements.primary.autocomplete('val', suggestion.primary_line);
-                config.elements.secondary.val('');
-                config.elements.city.val(suggestion.city);
-                config.elements.state.val(suggestion.state);
-                config.elements.zip.val(suggestion.zip_code);
+        /**
+         * Called after Lob verifies an address. Improves/updates and caches
+         * @param {object} data - verified address object returned from Lob
+         */
+        function fixAndSave(data) {
+          var didFix;
+          var parts = denormalizeParts(data, !!config.elements.secondary.val());
+          var address = config.address = {
+            primary: parts.primary_line,
+            secondary: parts.secondary_line,
+            city: data.components && data.components.city || '',
+            state: data.components && data.components.state || '',
+            zip: plus4(data.components)
+          }
+          for (var p in address) {
+            if (address.hasOwnProperty(p)) {
+              if (address[p].toUpperCase() != config.elements[p].val().toUpperCase() &&
+                !(p === 'zip' && config.elements[p].val().length === 5 && address[p].indexOf(config.elements[p].val()) == 0)) {
```

Comment:
`config.elements[p].val()) == 0`

Isn't that supposed to be triple equal?

Reply:
good catch